src/utility/StorageAPI.js

```diff
@@ -26,8 +26,10 @@ export default class StorageAPI {
     if (this.storage) {
       const key = `graphiql:${name}`;
       if (value) {
-        if (isStorageAvailable(this.storage, key, value)) {
```

Comment:
Clearly it is bad that it silently fails and that it literally writes the item already. However, could we maybe design a friendlier message? Some ideas:
* Tell the user to clear their local storage
* Tell the user that no storage is available and ask if we can clean up some old queries in his storage.

Reply:
I always assumed that the GraphiQL history was LRU; I think we should just delete old queries to make space for new ones (with the exception of starred queries, which should never be deleted). I don't think we need to ask permission. We should also probably have a byte-limit on the size of an actual item; if it's over, say, 100kB I don't think it should even attempt to be stored.